scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -79,17 +104,59 @@ impl PrecomputedQueries {
             .await
             .expect("Preparing delete query failed.");
 
+        let (c_ks_t, c_k_c) = (keyspace_table_name.clone(), keys_cond.clone());
+        let delete_value_queries = PreparedStatementCache {
+            queries: HashMap::new(),
+            // We have to use UPDATE with … = null syntax instead of a DELETE statement as
+            // DELETE will use incorrect timestamp for non-frozen collections. More info in the documentation:
+            // https://docs.scylladb.com/using-scylla/cdc/cdc-advanced-types/#collection-wide-tombstones-and-timestamps
+            query_string_maker: Box::new(move |column_name| format!(
+                "UPDATE {} SET {} = NULL WHERE {}",
+                c_ks_t,
+                column_name,
+                c_k_c,
+            ))
+        };
+
+        let (c_ks_t, c_k_c) = (keyspace_table_name.clone(), keys_cond.clone());
+        let overwrite_queries = PreparedStatementCache {
+            queries: HashMap::new(),
+            query_string_maker: Box::new(move |column_name| format!(
+                "UPDATE {} USING TTL ? SET {} = ? WHERE {}",
+                c_ks_t,
+                column_name,
+                c_k_c,
+            ))
+        };
+
+        let (c_ks_t, c_k_c) = (keyspace_table_name.clone(), keys_cond.clone());
+        let update_list_elements_queries = PreparedStatementCache {
+            queries: HashMap::new(),
+            query_string_maker: Box::new(move |column_name| format!(
+                "UPDATE {} USING TTL ? SET {}[SCYLLA_TIMEUUID_LIST_INDEX(?)] = ? WHERE {}",
+                c_ks_t,
+                column_name,
+                c_k_c,
+            ))
+        };
+
+        let (c_ks_t, c_k_c) = (keyspace_table_name.clone(), keys_cond.clone());
+        let update_map_or_set_elements_queries = PreparedStatementCache {
+            queries: HashMap::new(),
+            query_string_maker: Box::new(move |column_name| format!(
+                "UPDATE {tbl} USING TTL ? SET {cname} = {cname} + ?, {cname} = {cname} - ? WHERE {cond}",
+                tbl = c_ks_t,
+                cname = column_name,
+                cond = c_k_c,
+            )),
+        };
+
```

Comment:
Apparently, Rust does not have a convienient method for cloning items that we want to move into the closures: https://github.com/rust-lang/rfcs/issues/2407

Reply:
Yes, I'm afraid there is no better way to do this than to clone needed items before creating the closure.